### PR TITLE
Gea feature/search of equipment and medicine

### PR DESCRIPTION
**Pull request by stefanwert**

### CODE REVIEW

File: GraphicEditor/Repositories/EquipmentRepository.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            MySqlCommand sqlCommand = new MySqlCommand(query, connection);
+            MySqlDataReader sqlReader = sqlCommand.ExecuteReader();
+            List<Equipment> resultList = new List<Equipment>();
+            while (sqlReader.Read())
+            {
+                Equipment entity = new Equipment();
+                entity.Name = (string)sqlReader[2];
+                entity.Id = (string)sqlReader[0];
+                entity.RoomId = (string)sqlReader[3];
+                resultList.Add(entity);
```
Comment (Ana00000) [inline GraphicEditor/Repositories/EquipmentRepository.cs]: We have serial numbers and don't need ids.
Columns need to be filled in right way other way we will not get what we want for every attribute.

File: GraphicEditor/Repositories/MedicineRepository.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                MedicineGEA entity = new MedicineGEA();
+                entity.SerialNumber = (string)sqlReader[0];
+                entity.CopyrightName = (string)sqlReader[1];
+                entity.GenericName = (string)sqlReader[2];
+                entity.MedicineManufacturerId = (string)sqlReader[3];
+                entity.MedicineTypeId = (string)sqlReader[4];
+                resultList.Add(entity);
+            }
+            connection.Close();
+            return resultList;
```
Comment (Ana00000) [inline GraphicEditor/Repositories/MedicineRepository.cs:39]: You should use existent MedicineRepository from WebProject, not add another one...

File: GraphicEditor/View/Windows/EquipmentSearch.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        {
+            string resultOfSearch = equipmentName + " found at: ";
+            int equipmentCounter = equipments.Count;
+
+            if (equipmentCounter != 0)
+            {
+                int checkCounter = 0;
+                foreach (Equipment equipment in equipments)
+                {
+                    //roomRepository = new RoomRepository();
```
Comment (Ana00000) [inline GraphicEditor/View/Windows/EquipmentSearch.xaml.cs]: This should be deleted.

File: GraphicEditor/View/Windows/EquipmentSearch.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                    List<RoomGEA> rooms = roomRepository.GetRoomsBySerialNumber(equipment.RoomId);
+                    //dopuni !!!
```
Comment (Ana00000) [inline GraphicEditor/View/Windows/EquipmentSearch.xaml.cs]: This should be deleted.

File: WebApplication/Backend/Util/MyDbContext.cs [hunk 5]
```diff
@@ -20,6 +20,8 @@ public class MyDbContext : DbContext
         public DbSet<MedicineType> MedicineTypes { get; set; }
         public DbSet<Medicine> Medicine { get; set; }
 
+        public DbSet<MedicineGEA> MedicineGEA { get; set; }
```
Comment (Ana00000) [inline WebApplication/Backend/Util/MyDbContext.cs]: This is what i was saying in comment above. No need for MedicineGEA when Medicine is already created for need of gea team and it works fine.

File: WebApplication/Backend/Util/MyDbContext.cs [hunk 6]
```diff
@@ -33,6 +35,7 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
             modelBuilder.Entity<MedicineManufacturer>().HasKey(o => o.SerialNumber);
             modelBuilder.Entity<MedicineType>().HasKey(o => o.SerialNumber);
             modelBuilder.Entity<Medicine>().HasKey(o => o.SerialNumber);
+            modelBuilder.Entity<MedicineGEA>().HasKey(o => o.SerialNumber);
```
Comment (Ana00000) [inline WebApplication/Backend/Util/MyDbContext.cs:37]: No need.

File: WebApplication/Backend/Util/MyDbContext.cs [hunk 7]
```diff
@@ -64,6 +67,31 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
...
+                );
+            modelBuilder.Entity<MedicineType>().HasData(
+                    new MedicineType { SerialNumber = "11", Type = "Antibiotic" },
+                    new MedicineType { SerialNumber = "12", Type = "Brufen" }
+                );
+            modelBuilder.Entity<MedicineGEA>().HasData(
+                   new MedicineGEA { SerialNumber = "21", CopyrightName = "Brufen", GenericName = "Brufen", MedicineManufacturerId = "1", MedicineTypeId = "11" },
+                   new MedicineGEA { SerialNumber = "22", CopyrightName = "Probiotic", GenericName = "Probiotic", MedicineManufacturerId = "2", MedicineTypeId = "12" }
+                );
         }
```
Comment (Ana00000) [inline WebApplication/Backend/Util/MyDbContext.cs:94]: You are using data from created Medicine but creating MedicineGEA.

File: WebApplication/Migrations/20201128123201_init.Designer.cs [hunk 8]
```diff
@@ -0,0 +1,386 @@
+﻿// <auto-generated />
+using System;
```
Comment (Ana00000) [inline WebApplication/Migrations/20201128123201_init.Designer.cs]: Migration folder should be deleted.

File: project-generated-code-backend/HealthClinicBackend.csproj [hunk 9]
```diff
@@ -55,6 +55,7 @@
     <Compile Include="Backend\Model\Hospital\Building.cs" />
     <Compile Include="Backend\Model\Hospital\Floor.cs" />
     <Compile Include="Backend\Model\Hospital\Hospital.cs" />
+    <Compile Include="Backend\Model\Hospital\MedicineGEA.cs" />
```
Comment (Ana00000) [inline project-generated-code-backend/HealthClinicBackend.csproj:58]: No need.

File: project-generated-code-backend/Backend/Model/Hospital/MedicineGEA.cs [hunk 10]
```diff
@@ -0,0 +1,51 @@
+﻿using Backend.Model.Util;
```
Comment (Ana00000) [inline project-generated-code-backend/Backend/Model/Hospital/MedicineGEA.cs:1]: No need.

File: WebApplication/Migrations/20201129133215_num7.cs [hunk 11]
```diff
@@ -0,0 +1,41 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
```
Comment (Ana00000) [inline WebApplication/Migrations/20201129133215_num7.cs]: Delete migrations.

File: WebApplication/Migrations/20201129133041_num6.cs [hunk 12]
```diff
@@ -0,0 +1,41 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
```
Comment (Ana00000) [inline WebApplication/Migrations/20201129133041_num6.cs]: Delete migrations.

File: WebApplication/Migrations/20201129133041_num6.Designer.cs [hunk 13]
```diff
@@ -0,0 +1,522 @@
+﻿// <auto-generated />
```
Comment (Ana00000) [inline WebApplication/Migrations/20201129133041_num6.Designer.cs]: Delete migrations.

File: WebApplication/Migrations/20201129132937_num5.cs [hunk 14]
```diff
@@ -0,0 +1,106 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
```
Comment (Ana00000) [inline WebApplication/Migrations/20201129132937_num5.cs]: Delete migrations.

File: WebApplication/Migrations/20201129132937_num5.Designer.cs [hunk 15]
```diff
@@ -0,0 +1,522 @@
+﻿// <auto-generated />
```
Comment (Ana00000) [inline WebApplication/Migrations/20201129132937_num5.Designer.cs]: Delete migrations.

File: WebApplication/Migrations/20201128193328_num4.cs [hunk 16]
```diff
@@ -0,0 +1,102 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
```
Comment (Ana00000) [inline WebApplication/Migrations/20201128193328_num4.cs]: Delete migrations.

File: WebApplication/Migrations/20201128193328_num4.Designer.cs [hunk 17]
```diff
@@ -0,0 +1,458 @@
+﻿// <auto-generated />
```
Comment (Ana00000) [inline WebApplication/Migrations/20201128193328_num4.Designer.cs]: Delete migrations.

File: WebApplication/Migrations/20201128191014_num2.cs [hunk 18]
```diff
@@ -0,0 +1,55 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
```
Comment (Ana00000) [inline WebApplication/Migrations/20201128191014_num2.cs]: Delete migrations.

File: WebApplication/Migrations/20201128191014_num2.Designer.cs [hunk 19]
```diff
@@ -0,0 +1,395 @@
+﻿// <auto-generated />
```
Comment (Ana00000) [inline WebApplication/Migrations/20201128191014_num2.Designer.cs]: Delete migrations.

File: WebApplication/Migrations/20201128185654_num1.cs [hunk 20]
```diff
@@ -0,0 +1,51 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
```
Comment (Ana00000) [inline WebApplication/Migrations/20201128185654_num1.cs]: Delete migrations.

File: WebApplication/Migrations/20201128185654_num1.Designer.cs [hunk 21]
```diff
@@ -0,0 +1,395 @@
+﻿// <auto-generated />
```
Comment (Ana00000) [inline WebApplication/Migrations/20201128185654_num1.Designer.cs]: Delete migrations.

File: WebApplication/Migrations/20201128123201_init.cs [hunk 22]
```diff
@@ -0,0 +1,242 @@
+﻿using System;
```
Comment (Ana00000) [inline WebApplication/Migrations/20201128123201_init.cs]: Delete migrations.

File: GraphicEditor/Repositories/EquipmentRepository.cs (hunk 1 above)
Comment (Ana00000) [reply GraphicEditor/Repositories/EquipmentRepository.cs]: Equipment is inheriting Entity, which contain SerialNumber. Serial number will be placed at sqlReader[0] by default so you'll get serial number for id. Change id to serial number here and remove ids from model of equipment. This way everything will be in order and than you can put name on [1] and roomId on [2].

File: GraphicEditor/Repositories/MedicineRepository.cs (hunk 2 above)
Comment (stefanwert) [reply GraphicEditor/Repositories/MedicineRepository.cs:39]: For now i will use another one because i can't reference on web project from GEA.

Comment (milicasiriski) [reply GraphicEditor/Repositories/MedicineRepository.cs:39]: Hm, is this still a problem with .NET versions? Can you try to make Web Repository and IRepository public and try that way?

File: GraphicEditorTests/EquipmentSearchesTests.cs [hunk 23]
```diff
@@ -0,0 +1,41 @@
...
+            var equipments = _eqipmentRepository.GetEquipmentsByName("Bed");
+
+            //Assert
+            Assert.NotNull(equipments);
+            foreach (var equipment in equipments)
+                Assert.Equal("Bed", equipment.Name);
+        }
+
+        [Fact]
+        public void GetEquipmentByName_EquipmentExist_ReturnNull()
```
Comment (milicasiriski) [inline GraphicEditorTests/EquipmentSearchesTests.cs]: EquipmentDoesNotExist maybe?

**Review by milicasiriski — CHANGES_REQUESTED** [review]
@Ana00000 already covered most changes 😄 Just please delete migrations. Tests are great!

File: GraphicEditor/Repositories/MedicineRepository.cs (hunk 2 above)
Comment (stefanwert) [reply GraphicEditor/Repositories/MedicineRepository.cs:39]: Yes , we still have that problem . However we will solve it in future.

File: GraphicEditor/Repositories/Interfaces/IEqipmentRepository.cs [hunk 24]
```diff
@@ -0,0 +1,15 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GraphicEditor.Repositories.Interfaces
+{
+    public interface IEqipmentRepository
+    {
```
Comment (lemara98) [inline GraphicEditor/Repositories/Interfaces/IEqipmentRepository.cs]: Name of the file is incorrect. This is just a note.

File: GraphicEditor/View/Windows/EquipmentSearch.xaml [hunk 25]
```diff
@@ -0,0 +1,51 @@
...
+        <Button Content="Search"
+               Click="SearchEquipmentOrMedicine"
+                HorizontalAlignment="Left"
+                Height="34"
+                VerticalAlignment="Top"
+                Width="88" 
+                Margin="463,54,0,0" />
+        <TextBlock x:Name="SearcheEqipmentTextBlock" 
+                   HorizontalAlignment="Left" 
+                   Height="216"
```
Comment (lemara98) [inline GraphicEditor/View/Windows/EquipmentSearch.xaml]: x:Name="SearcheEqipmentTextBlock"  spelling

**Review by lemara98 — APPROVED** [review]

File: project-generated-code-backend/Backend/Model/Hospital/Building.cs [hunk 26]
```diff
@@ -5,17 +5,13 @@ namespace health_clinic_class_diagram.Backend.Model.Hospital
 {
     public class Building : Entity
     {
-        public string Id { get; set; }
         public string Name { get; set; }
-        public List<Floor> Floors { get; set; }
```
Comment (Ana00000) [inline project-generated-code-backend/Backend/Model/Hospital/Building.cs:10]: Nice job. :))

File: project-generated-code-backend/Backend/Model/Hospital/Building.cs [hunk 27]
```diff
@@ -5,17 +5,13 @@ namespace health_clinic_class_diagram.Backend.Model.Hospital
 {
     public class Building : Entity
     {
-        public string Id { get; set; }
```
Comment (Ana00000) [inline project-generated-code-backend/Backend/Model/Hospital/Building.cs:8]: Now it's perfect. 🥇

**Comment by Ana00000** [comment]
[(2) Search for equipment and medicine](https://trello.com/c/ZAcAVfYk/49-2-search-for-equipment-and-medicine)

**Review by Ana00000 — APPROVED** [review]

**Review by milicasiriski — APPROVED** [review]

**Review by lukaDoric — APPROVED** [review]

**Comment by stefanwert** [comment]
Search of equipment and medicine.

**Pull request merged**